tests/system-tests/rdscore/internal/rdscorecommon/sriov-rootless-dpdk.go

```diff
@@ -0,0 +1,861 @@
+				interfaceName, clientPod.Definition.Name, clientPod.Definition.Namespace, err)
+
+			return fmt.Errorf("failed to get link %s info from pod %s in namespace %s with error %w",
+				interfaceName, clientPod.Definition.Name, clientPod.Definition.Namespace, err)
+		}
+
+		glog.V(100).Infof("Current Rate for the interface %s: %d", interfaceName, currentRate)
+
+		currentRate -= originalRate
+
+		glog.V(100).Infof("Current run Rate for the interface %s: %d", interfaceName, currentRate)
+
+		if TrafficRate == 5000 && currentRate > TrafficRate || TrafficRate != 5000 && currentRate < TrafficRate {
+			glog.V(100).Infof("Failed traffic rate is not in expected range; "+
+				"current rate is %d, TrafficRate is %d, interfaceName is %s", currentRate, TrafficRate, interfaceName)
+
+			return fmt.Errorf("failed traffic rate is not in expected range; "+
+				"current rate is %d, TrafficRate is %d, interfaceName is %s", currentRate, TrafficRate, interfaceName)
+		}
+	}
+
+	return nil
+}
+
+func getLinkRx(runningPod *pod.Builder, linkName string) (int, error) {
+	var (
+		linkRawInfo bytes.Buffer
+		err         error
+	)
+
+	err = wait.PollUntilContextTimeout(
+		context.TODO(),
+		time.Second,
+		time.Second*30,
+		true,
+		func(ctx context.Context) (bool, error) {
+			linkRawInfo, err = runningPod.ExecCommand(
+				[]string{"/bin/bash", "-c", fmt.Sprintf("ip --json -s link show dev %s", linkName)})
+			if err != nil {
+				glog.V(100).Infof("The link %s info is not available for the pod %s in namespace %s "+
+					"with error %v",
+					linkName, runningPod.Definition.Name, runningPod.Definition.Namespace, err)
+
+				return false, nil
+			}
+
+			return true, nil
+		})
+
+	if err != nil {
+		glog.V(100).Infof("Failed to get link %s info from pod %s in namespace %s with error %v",
+			linkName, runningPod.Definition.Name, runningPod.Definition.Namespace, err)
+
+		return 0, fmt.Errorf("failed to get link %s info from pod %s in namespace %s with error %w",
+			linkName, runningPod.Definition.Name, runningPod.Definition.Namespace, err)
+	}
+
+	linkInfo, err := link.NewBuilder(linkRawInfo)
+	if err != nil {
+		glog.V(100).Infof("Failed to collect link %s info from pod %s in namespace %s with error %v",
+			linkName, runningPod.Definition.Name, runningPod.Definition.Namespace, err)
+
+		return 0, fmt.Errorf("failed to collect link %s info from pod %s in namespace %s with error %w",
+			linkName, runningPod.Definition.Name, runningPod.Definition.Namespace, err)
+	}
+
+	return linkInfo.GetRxByte(), nil
+}
+
+func getPCIAddressListFromSrIovNetworkName(podNetworkStatus string) ([]string, error) {
+	var podNetworkStatusType []podNetworkAnnotation
+	err := json.Unmarshal([]byte(podNetworkStatus), &podNetworkStatusType)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var pciAddressList []string
+
+	for _, networkAnnotation := range podNetworkStatusType {
+		if strings.Contains(networkAnnotation.Name, dpdkNetworkOne) {
```

Comment:
to make the method universal I suggest making `dpdkNetworkOne` as a parameter to `getPCIAddressListFromSrIovNetworkName`